pkg/deployments/src/test.ts

```diff
@@ -0,0 +1,58 @@
+import { RunSuperFunction, HardhatRuntimeEnvironment, HttpNetworkConfig, HardhatNetworkConfig } from 'hardhat/types';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+
+export default async function (args: any, hre: HardhatRuntimeEnvironment, run: RunSuperFunction<any>): Promise<void> {
+  if (hre.network.name === 'hardhat' && !args.fork) await runNormalTests(args, hre, run);
+  else if (hre.network.name === 'hardhat' && args.fork) await runForkTests(args, hre, run);
+  else await runDeployTests(args, hre, run);
+}
+
+async function runNormalTests(args: any, hre: HardhatRuntimeEnvironment, run: RunSuperFunction<any>): Promise<void> {
+  console.log('Running normal tests...');
+  args.testFiles = args.testFiles.filter((file: string) => file.endsWith('.test.ts'));
+  await run(args);
+}
+
+async function runDeployTests(args: any, hre: HardhatRuntimeEnvironment, run: RunSuperFunction<any>): Promise<void> {
+  console.log('Running deployment tests...');
+  if (args.fork) throw Error('Cannot test forks from non-local networks');
+  args.testFiles = args.testFiles.filter((file: string) => file.endsWith('.deploy.ts'));
+  await run(args);
+}
+
+async function runForkTests(args: any, hre: HardhatRuntimeEnvironment, run: RunSuperFunction<any>): Promise<void> {
+  console.log('Running fork tests...');
+  if (args.fork === 'hardhat') throw Error('Cannot test forks using local networks');
```

Comment:
```suggestion
  if (args.fork === 'hardhat') throw Error('Cannot fork local networks');
```

I'd argue fork tests _do_ use local networks :stuck_out_tongue: it's just that we don't fork them